react/DatePicker/index.jsx

```diff
@@ -0,0 +1,302 @@
+import LocaleFR from 'date-fns/locale/fr'
+import subDays from 'date-fns/subDays'
+import PropTypes from 'prop-types'
+import React, { forwardRef, useState } from 'react'
+
+import { makeFormat } from './helpers'
+import withOwnLocales from './locales/withOwnLocales'
+import useBreakpoints from '../providers/Breakpoints'
+import { useI18n } from '../providers/I18n'
+import { makeStyles } from '../styles'
+
+const localesFNS = {
+  fr: LocaleFR,
+  en: LocaleEN
+}
+
+const useStyles = makeStyles(() => ({
+  overrides: {
+    width: '100%',
+    height: isDesktop => (isDesktop ? '5rem' : 'inehrit'),
+    MuiOutlinedInput: {
+      '&:focused': {
+        notchedOutline: {
+          borderColor: 'var(--primaryColor)'
+        }
+      }
+    }
+  }
+}))
+
+const DatePicker = forwardRef(
+  (
+    {
+      className,
+      label,
+      clearable = false,
+      value = null,
+      placeholder,
+      isValid,
+      onFocus,
+      onBlur,
+      onChange,
+      minDate,
+      minDateLabelError,
+      format,
+      cancelLabel,
+      clearLabel,
+      okLabel,
+      todayLabel,
+      showTodayButton = false,
+      helperText,
+      errorLabel,
+      inputVariant = 'outlined',
+      inputProps,
+      KeyboardButtonProps,
+      enableKeyboard,
+      mode = 'date',
+      ampm,
+      ...props
+    },
+    ref
+  ) => {
+    const [error, setError] = useState(null)
+    const [isFocused, setIsFocused] = useState(false)
+
+    const { isDesktop } = useBreakpoints()
+    const classes = useStyles(isDesktop)
+    const { t, lang } = useI18n()
+
+    const isError = !isFocused && Boolean(error)
+    const _helperText = isError ? error : helperText ?? null
+    const _format = format || makeFormat({ ampm, mode, lang })
+    const _ampm = ampm ?? !(mode === 'time' || mode === 'dateTime')
+    const _placeholder = placeholder ?? formatFNS(new Date(), _format)
+    const _clearLabel = clearLabel || t('clear')
+    const _todayLabel = todayLabel || t('today')
+    const _cancelLabel = cancelLabel || t('cancel')
+    const _okLabel = okLabel || t('ok')
+    const _minDateLabelError = minDateLabelError
+      ? minDateLabelError
+      : minDate
+      ? t('minDateLabelError', {
+          date: formatFNS(minDate, _format)
+        })
+      : null
+
+    const _KeyboardButtonProps = {
+      'aria-label': label,
+      ...KeyboardButtonProps
+    }
+    const _inputProps = { inputMode: 'numeric', ...inputProps }
+
+    const handleChange = val => {
+      if (val?.toString() !== 'Invalid Date') {
+        if (minDate && isBefore(val, subDays(minDate, 1))) {
+          isValid?.(false)
+          setError(_minDateLabelError)
+          return
+        }
+        setError(null)
+        isValid?.(true)
+        onChange(val)
+      } else if (val === '') {
+        setError(null)
+        isValid?.(true)
+        onChange(null)
+      } else {
+        setError(errorLabel ?? t('invalidDate'))
+        isValid?.(false)
```

Comment:
What is the purpose of `isValid` ? Is it a callback ?

Reply:
Yes, for example, in our apps we very often use modals, and in this case we also want to make the modal buttons dynamic.
This callback is used to obtain the validity or invalidity of the field in order to pass it on to other components.

Reply:
Ok 👍 nit: Is it confusing to have a callback that does not start with `on`.

Reply:
Indeed, I don't like this approach either.
We opted for a second parameter in the `onChange` callback.